pkg/provider/data_sync.go

```diff
@@ -64,6 +64,12 @@ func DataSync() *schema.Resource {
 				Optional:    true,
 				Default:     syncDefaults.Branch,
 			},
+			"secret": {
+				Description: "The name of the secret that is referenced by GitRepository as SecretRef.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
```

Comment:
I would ranter have that we stick to using the option defaults rather than making this a computed value.

Reply:
I see, that was also my first idea. (more on the compat issue below)